src/components/legend/draw.js

```diff
@@ -590,14 +591,14 @@ function computeLegendDimensions(gd, groups, traces) {
             maxTraceWidth = 0,
             offsetX = 0;
 
-        // calculate largest width for traces and use for width of all legend items
+        // calculate largest width for traces and use for width of all legend items (if horizontalspacing mode is 'column')
         traces.each(function(d) {
```

Comment:
`maxTraceWidth` isn't used unless `isHorizontalColumn` is `true`, so don't compute it if not.